poshc2/server/Payloads.py

```diff
@@ -535,8 +532,16 @@ def CreateEXEFiles(self, sourcefile, payloadtype, name=""):
             f.write(content)
 
         # Compile the exe
-        subprocess.check_output("x86_64-w64-mingw32-gcc -w %s%s%s_%s_x64.c -o %s%s%s_%s_x64.exe" % (self.BaseDirectory, name, payloadtype.value, sourcefile.replace(".c",""), self.BaseDirectory, name, payloadtype.value, sourcefile.replace(".c","")), shell=True)
-        subprocess.check_output("i686-w64-mingw32-gcc -w %s%s%s_%s_x86.c -o %s%s%s_%s_x86.exe" % (self.BaseDirectory, name, payloadtype.value, sourcefile.replace(".c",""), self.BaseDirectory, name, payloadtype.value, sourcefile.replace(".c","")), shell=True)
+        if "dllmain" in content.lower():
```

Comment:
Is there another way to check if it's a DLL? 

We're checking if `dllmain` is in `content`, which is the full source of the payload template, so even if there's just a comment with dllmain in it it'll trigger this again.

Anything where we load user controlled input I'd try and avoid checks like this if possible...